### ISSUE TITLE
onPaymentMethodReceived got fired multiple times

### ISSUE THREAD

**Issue by gaugau** [jeffcarp/braintree-react#26]
Hi,

I've remounted the DropIn component after the transaction got rejected by the backend. But it seems that in the next submit, the onPaymentMethodReceived has been triggered twice, and keep increasing after each `tearnDown`. I'm not sure if it's a problem with the react component or braintree-web itself. TIA

```javascript
handleFailed() {
   setState({dropInKey: 'new key to force remounting'});
}
render() {
          <DropIn
            key={this.state.dropInKey}
            braintree={braintree}
            clientToken={tokenFetch.value.token}
            onReady={this.onReady}
            onError={this.onError}
            onPaymentMethodReceived={this.onPaymentMethodReceived}
          />
}
```

**Comment by jeffcarp** [comment]
This is probably a problem with braintree-react. We are probably not using teardown properly.

In the short term, would hiding Drop-In using `display: none;` instead of removing it from the DOM fix the problem?

**Comment by gaugau** [comment]
@jeffcarp Nope,  I need to completely remount the component (`teardown`) in order to generate a new nonce. Otherwise, it keeps generating the same `payment token nonce` and the backend could not process it. :(

**Comment by jeffcarp** [comment]
Ok, I'm on vacation right now but I can look into what's causing this when I get back.

**Comment by gaugau** [comment]
@jeffcarp Thanks, I'll try my best to find a workaround. Enjoy your vacation ;)

**Comment by jeffcarp** [comment]
It's been a while and I still need to try to replicate this. Ideally we should have a test to make sure `teardown` is working correctly.

**Comment by rmdydo** [comment]
I am having the same issue as @gaugau, it appears that the teardown method is not removing the submit handlers on the containing form and subsequent initializations just add more handlers. @gaugau did you figure out any solutions for this?
